tests/phpunit/tests/includes/collection/class-test-posts.php

```diff
@@ -1343,6 +1343,105 @@ public function test_remove_hashtags_preserves_structure() {
 		$this->assertStringNotContainsString( '#php', $result );
 	}
 
+	/**
+	 * Test delete_all method deletes all posts.
+	 *
+	 * @covers ::delete_all
+	 */
+	public function test_delete_all() {
+		// Create some posts.
+		self::factory()->post->create_many(
+			5,
+			array(
+				'post_type'   => Posts::POST_TYPE,
+				'post_status' => 'publish',
+			)
+		);
+
+		// Verify posts were created.
+		$count_before = \wp_count_posts( Posts::POST_TYPE )->publish;
+		$this->assertEquals( 5, $count_before );
+
+		// Delete all posts.
+		$deleted = Posts::delete_all();
+
+		// Clear cache to get accurate count.
+		\wp_cache_delete( \_count_posts_cache_key( Posts::POST_TYPE ), 'counts' );
```

Comment:
Is this necessary? I'd expect `wp_delete_post()` to take care of it.

If it is necessary, we should probably clear the post count cache in `Posts::delete_all()`.

Reply:
Let's move the cache clearing into `Posts::delete_all()` then? It shouldn't be up to the calling code to know that.